src/Reflection/ReflectionAttribute.php

```diff
@@ -11,19 +11,31 @@
 use Roave\BetterReflection\Reflection\StringCast\ReflectionAttributeStringCast;
 use Roave\BetterReflection\Reflector\Reflector;
 
+use function array_map;
+
 class ReflectionAttribute
 {
+    private string $name;
+
+    /** @var array<int|string, Node\Expr> */
+    private array $arguments = [];
+
     public function __construct(
         private Reflector $reflector,
-        private Node\Attribute $node,
+        Node\Attribute $node,
         private ReflectionClass|ReflectionMethod|ReflectionFunction|ReflectionClassConstant|ReflectionEnumCase|ReflectionProperty|ReflectionParameter $owner,
         private bool $isRepeated,
     ) {
+        $this->name = $node->name->toString();
+
+        foreach ($node->args as $argNo => $arg) {
+            $this->arguments[$arg->name?->toString() ?? $argNo] = $arg->value;
+        }
     }
 
     public function getName(): string
```

Comment:
Can we somehow declare `non-empty-string` from here?

Reply:
Done.